### PR TITLE
Use reflection to support both versions of Android Studio

### LINKED ISSUE #4: incompatibility with Gradle 5.1
When using the AndroidStudio 3.4 Canary 10,
and having upgraded to Gradle 5.1-milestone-1,

the following error appears while building:

```
Caused by: java.lang.NoSuchMethodError: com.android.build.gradle.tasks.MergeSourceSetFolders.getOutputDir()Ljava/io/File;
	at com.fourlastor.pickle.PicklePlugin$configure$1.execute(PicklePlugin.kt:115)
	at com.fourlastor.pickle.PicklePlugin$configure$1.execute(PicklePlugin.kt:23)
	at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:158)
	at com.fourlastor.pickle.PicklePlugin.configure(PicklePlugin.kt:109)
	at com.fourlastor.pickle.PicklePlugin.access$configure(PicklePlugin.kt:23)
	at com.fourlastor.pickle.PicklePlugin$apply$1$1.execute(PicklePlugin.kt:96)
	at com.fourlastor.pickle.PicklePlugin$apply$1$1.execute(PicklePlugin.kt:23)
	at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:158)
	at com.fourlastor.pickle.PicklePlugin$apply$1.execute(PicklePlugin.kt:87)
	at com.fourlastor.pickle.PicklePlugin$apply$1.execute(PicklePlugin.kt:23)

```

### PR DESCRIPTION
**Pull request by tasomaniac**
Android Studio did a break change with the public task API on Android Studio 3.4.

This breaks the compatibility between Pickle and AS 3.4

If the plugin would've been in Groovy, this would be 1 line fix. Kotlin is not so dynamic :trollface: 

Anyways, here is a fix to support all versions. It uses reflection utilities from Gradle to access new APIs. If they're not available, it fallbacks to original. 

Fixes #4

### CODE REVIEW

File: plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt [hunk 1]
```diff
@@ -87,12 +91,12 @@ class PicklePlugin : Plugin<Project> {
             project.plugins.all {
                 when (it) {
                     is LibraryPlugin -> {
-                        project.extensions.findByType(LibraryExtension::class.java).run {
+                        project.extensions.findByType(LibraryExtension::class.java)?.run {
```
Comment (fourlastor) [inline plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt]: why the `?` here?

File: plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt [hunk 2]
```diff
@@ -87,12 +91,12 @@ class PicklePlugin : Plugin<Project> {
...
                             configure(project, testVariants, extension)
                         }
                     }
                     is AppPlugin -> {
-                        project.extensions.findByType(AppExtension::class.java).run {
+                        project.extensions.findByType(AppExtension::class.java)?.run {
```
Comment (fourlastor) [inline plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt]: why the `?` here?

File: plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt [hunk 3]
```diff
@@ -121,4 +125,15 @@ class PicklePlugin : Plugin<Project> {
...
         }
     }
+
+    fun TestVariant.resolveOutputDir(): File {
+        return try {
+            val taskProvider = method(this, Any::class.java, "getMergeAssetsProvider").invoke(this)
+            val mergeAssets = method(taskProvider, MergeSourceSetFolders::class.java, "get").invoke(taskProvider)
+            val outputDirProvider = method(mergeAssets, Provider::class.java, "getOutputDir").invoke(mergeAssets)
+            (outputDirProvider.get() as Directory).asFile
+        } catch (ignored: Exception) {
```
Comment (fourlastor) [inline plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt]: can we have a more specific catched exception here?

File: plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt (hunk 1 above)
Comment (tasomaniac) [reply plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt]: I was testing with the newer Android Plugin. findByType is nullable now which makes sense. I decided to keep it.

Comment (tasomaniac) [reply plugin/src/main/java/com/fourlastor/pickle/PicklePlugin.kt]: Just removed it. It's not so necessary atm

**Comment by fourlastor** [comment]
Thank you @tasomaniac!

**Pull request merged**